src/production/forms.py

```diff
@@ -5,10 +5,63 @@
 
 from django import forms
 
-from production import models
+from production import models, logic
 
 
 class TypesetterNote(forms.ModelForm):
     class Meta:
         model = models.TypesetTask
         fields = ('note_from_typesetter',)
+
+
+class AssignTypesetter(forms.ModelForm):
+
+    def __init__(self, *args, **kwargs):
+        self.files = kwargs.pop('files')
+        self.article = kwargs.pop('article')
+        self.typesetters = kwargs.pop('typesetters')
+        self.assignment = kwargs.pop('assignment')
+
+        super(AssignTypesetter, self).__init__(*args, **kwargs)
+
+    class Meta:
+        model = models.TypesetTask
+        fields = (
+            'typesetter',
+            'files_for_typesetting',
+            'typeset_task',
+        )
+
+    def clean(self):
+        cleaned_data = super().clean()
+
+        file_check = logic.check_posted_typesetter_files(
+            self.article,
+            self.files,
+            cleaned_data.get('files_for_typesetting'),
+        )
+
+        if not file_check:
+            self.add_error(
+                'files_for_typesetting',
+                'File not found in files list.'
+            )
+
+        if not cleaned_data.get('typesetter') in logic.typesetter_users(
```

Comment:
Just curious, how can we end up with a typesettter who is not a typesetter here?

Reply:
If someone attempted to change the ID of one of the radio buttons. We want to check that the POST'd ID hasn't been tampered with.